### PR TITLE
feat: contact support dialog

### PR DESCRIPTION
**Pull request by jhasuraj01**

### CODE REVIEW

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  ContactSupportDialogProviderProps
+> = ({ children }) => {
+  const lang = useAppSelector(selectLanguage);
+  const dispatch = useAppDispatch();
+  const deviceRequiredDialogsMap: Record<number, number[] | undefined> = {};
+
+  const categories: DropDownListItemProps[] = [
+    { text: 'Feedback', id: 'feedback' },
+    { text: 'Bug Report', id: 'bugReport' },
+  ];
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: These should be the categories. 
```
'Complaint', 'Feedback', 'Others'
```

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+  useEffect(validateForm, [
+    isLoading,
+    isEmailError,
+    email,
+    isCategoryError,
+    selectedCategory,
+    isDescriptionError,
+    description,
+  ]);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: Use `useMemo`

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  };
+
+  const handleContactSupportSubmit = async () => {
+    setIsLoading(true);
+    setError(null);
+
+    const category = categories.find(c => c.id === selectedCategory);
+
+    if (email === null || description === null || category === undefined)
+      return;
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]:
```suggestion
    if (email === null || description === null || category === undefined) {
      setIsLoading(false);
      return;
    }
```

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+      setError(
+        ((apiError as AxiosError).response?.data as string) ??
+          (apiError as AxiosError).message,
+      );
+      setIsLoading(false);
+      return;
+    }
+
+    setIsLoading(false);
+    onNext();
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]:
```suggestion
    try {
      await sendFeedback(email, category.text, description);
      setIsLoading(false);
      onNext();
    } catch (apiError) {
      setError(
        ((apiError as AxiosError).response?.data as string) ??
          (apiError as AxiosError).message,
      );
      setIsLoading(false);
    }
```

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+  };
+
+  const handleCategorySelection = (id: string | undefined) => {
+    setSelectedCategory(id);
+  };
+
+  const tabs: ITabs = [
+    {
+      name: lang.strings.dialogs.password.confimPassword.title,
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: Incorrect name

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: updated categories

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx (hunk 3 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: done

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx (hunk 4 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: done

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx (hunk 5 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: name corrected

File: packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx (hunk 2 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/ContactSupport/context/index.tsx]: replaced useEffect & setState with useMemo

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**